K8s-deployment/K8s-cluster/addons/velero/azure/README.md

```diff
@@ -0,0 +1,228 @@
+## Description
+[**Velero**](https://velero.io/docs/v1.6/) is the tool being used for backing up and restoring the Kubernetes cluster resources. It is being used to backup persistent volumes and persistent volume claims for the following deployments:
+
+- Redis-cluster
+- PostgreSQL
+- ImmuDB
+- RabbitMQ
+### Deployment Architecture
+Velero consists of:
+- A server that runs on your kubernetes cluster.
+- A command-line client that runs locally on the bootstrap machine (Rancher server node).
+#### Cloud and OSS components
+- volume snapshots stored in Azure Blob storage.
+
+## Deployment
+### Pre-requisites
+
+- Access to a Kubernetes cluster, v1.12 or later, with DNS and container networking enabled.
+- kubectl installed locally.
+
+### Deploy
+If you do not have the `az` Azure CLI 2.0 installed locally, follow the [instruction](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) to set it up.
+Run:
+
+```bash
+az login
+```
+#### Setup Azure storage account and blob container
+
+Velero requires a storage account and blob container in which to store backups. The storage account needs to be created with a globally unique id since this is used for dns.
+
+Create Azure storage account and blob container, replacing placeholders appropriately:
+
+```bash
+AZURE_BACKUP_RESOURCE_GROUP=<YOUR_RESOURCE_GROUP>
+AZURE_STORAGE_ACCOUNT_ID=<UNIQUE_NAME> 
+//AZURE_STORAGE_ACCOUNT_ID="velero$(uuidgen | cut -d '-' -f5 | tr '[A-Z]' '[a-z]')" for Unique Id
+
+az storage account create \
+    --name $AZURE_STORAGE_ACCOUNT_ID \
+    --resource-group $AZURE_BACKUP_RESOURCE_GROUP \
+    --sku Standard_GRS \
+    --encryption-services blob \
+    --https-only true \
+    --kind BlobStorage \
+    --access-tier Hot
+```
+Create the blob container named velero. Feel free to use a different name, preferably unique to a single Kubernetes cluster.
+
+```bash
+BLOB_CONTAINER=<YOUR_CONTAINER_NAME>
+az storage container create -n $BLOB_CONTAINER --public-access off --account-name $AZURE_STORAGE_ACCOUNT_ID
+```
+
+
+
+#### Set permissions for Velero
+Set the name of the Resource Group that contains your Kubernetes cluster's virtual machines/disks.
+
+```bash
+    AZURE_RESOURCE_GROUP=<NAME_OF_RESOURCE_GROUP>
+ ```
+Obtain your Azure Account Subscription ID:
+   ```
+   AZURE_SUBSCRIPTION_ID=`az account list --query '[?isDefault].id' -o tsv`
+   ```
+##### Set permissions using service principal
+There are several ways Velero can authenticate to Azure: (1) by using a Velero-specific service principal (2) by using AAD Pod Identity or (3) by using a storage account access key.
```

Comment:
Mention that we'll be using the first one for our deployment so that it's clear, Since only that is present in this doc. And maybe just add a link to the other methods if you're mentioning them.